platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -54,15 +54,14 @@ public class DefaultProblemBuilder implements InternalProblemBuilder {
     private ProblemId id;
     private String contextualLabel;
     private Severity severity;
-    private final ImmutableList.Builder<ProblemLocation> locations = ImmutableList.builder();
-    private final ImmutableList.Builder<ProblemLocation> contextLocations = ImmutableList.builder();
+    private final List<ProblemLocation> locations = new ArrayList<ProblemLocation>();
```

Comment:
`List` to be able to use `contains` to check whether a location is already contained.